### PR TITLE
make eventHooks thread safe

### PR DESCRIPTION
**Pull request by tw4452852**
Signed-off-by: Tw <tw19881113@gmail.com>

### 1. What does this change do, exactly?

Use `sync.Map` to make `eventHooks` concurrent r/w safe

### 2. Please link to the relevant issues.

#1974 

### 3. Which documentation changes (if any) need to be made because of this PR?
N/A

### 4. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

**Comment by tw4452852** [comment]
Hi @mholt,

I think the main advantage of `sync.Map` is that it doesn't incur the blocking while `sync.Mutex` does.
Because `sync.Map` use atomic operation underneath. In the case of #1974 , the signal handle may wait  until the plugin initializes done if we use a mutex.

**Comment by mholt** [comment]
For the record: This change will require Caddy to be built using Go 1.9 instead of 1.8. Which I suppose is fine, with Go 1.10 around the corner.

File: plugins.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ var (
 
 	// eventHooks is a map of hook name to Hook. All hooks plugins
 	// must have a name.
-	eventHooks = make(map[string]EventHook)
+	eventHooks = sync.Map{}
```
Comment (mholt) [inline plugins.go]: Should this be a pointer 🤔 ? I know we don't pass this value around, but [all the methods are defined on `*sync.Map`](https://golang.org/pkg/sync/#Map). (I know it says the zero value is valid, but I just want to double-check.)

Comment (tw4452852) [reply plugins.go]: No, this isn't necessary, please check this:
https://stackoverflow.com/a/47825590

**Comment by mholt** [comment]
Thanks as always, Tw!

**Pull request merged**